.github/workflows/ci-kafka.yml

```diff
@@ -27,7 +27,17 @@ jobs:
         go-version: 1.20.x
 
     - name: Run kafka integration tests
-      run: bash scripts/kafka-integration-test.sh
+      run: |
+            retrycount=0
+             maxRetries=3
+             backoff=1
+             while[$retrycount -lt $maxRetries];do
+              bash scripts/kafka-integration-test.sh && break
+             echo "Tests failed, retrying in $backoff seconds..."
+             sleep $backoff
```

Comment:
backoff logic is largely irrelevant, you go from 1s -> 2s -> 4s, while the actual test runs much longer